managed/services/agents/roster.go

```diff
@@ -35,36 +39,68 @@ const (
 type roster struct {
 	l *logrus.Entry
 
+	db *reform.DB
 	rw sync.RWMutex
 	m  map[string][]string
 }
 
-func newRoster() *roster {
+// newRoster creates a new roster instance.
+func newRoster(db *reform.DB) *roster {
 	return &roster{
-		l: logrus.WithField("component", "roster"),
-		m: make(map[string][]string),
+		db: db,
+		l:  logrus.WithField("component", "roster"),
+		m:  make(map[string][]string),
 	}
 }
 
-func (r *roster) add(pmmAgentID string, group agentGroup, agentIDs []string) (groupID string) { //nolint:nonamedreturns
+// add adds a new group of exporter IDs to the roster.
+func (r *roster) add(pmmAgentID string, group agentGroup, exporters map[*models.Node]*models.Agent) string {
 	r.rw.Lock()
 	defer r.rw.Unlock()
 
-	groupID = pmmAgentID + "/" + string(group)
-	r.m[groupID] = agentIDs
-	r.l.Debugf("add: %s = %v", groupID, agentIDs)
-	return
+	groupID := pmmAgentID + "/" + string(group)
+	exporterIDs := make([]string, 0, len(exporters))
+	for _, exporter := range exporters {
+		exporterIDs = append(exporterIDs, exporter.AgentID)
+	}
+
+	sort.Strings(exporterIDs)
+
+	r.m[groupID] = exporterIDs
+	r.l.Debugf("add: %s = %v", groupID, exporterIDs)
+	return groupID
 }
 
-func (r *roster) get(groupID string) (agentIDs []string) { //nolint:nonamedreturns
+// get returns a PMMAgentID and a group of exporter IDs for a given Group ID.
+func (r *roster) get(groupID string) (string, []string, error) {
 	r.rw.RLock()
 	defer r.rw.RUnlock()
 
-	agentIDs = r.m[groupID]
+	suffix := "/" + string(rdsGroup)
+	PMMAgentID, ok := strings.CutSuffix(groupID, suffix)
```

Comment:
I think this is not critical, a matter of taste, but it looks like we don’t need to allocate memory for a `suffix` every time and we could create it once using const at the compile time with `const suffix = "/" + string(rdsGroup)` [This](https://stackoverflow.com/questions/35026810/runtime-const-in-golang) place describes this minor tweak.

Reply:
No problem, done!